src/components/Icon/Icon.tsx

```diff
@@ -21,6 +22,29 @@ interface Props extends IconProps {}
 
 export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
   const i18n = useI18n();
+  const unstableMonorail = useMonorail();
+
+  /* eslint-disable babel/camelcase */
+  useEffect(() => {
+    if (unstableMonorail) {
+      let loggedSource;
+      if (typeof source === 'function') {
```

Comment:
nitpick: I'd be tempted to the "determine the loggedSource based on the source" logic out into a separate function